test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -1587,6 +1588,118 @@ public async Task RestoreCommand_CentralVersion_ErrorWhenCentralPackageVersionFi
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                pathContext.PackageSource,
+                PackageSaveMode.Defaultv3,
+                    new SimpleTestPackageContext("foo", "1.0.0"));
+
+                using var context = new SourceCacheContext();
+
+                var packageSources = new List<PackageSource>
+                {
+                    new PackageSource(pathContext.PackageSource),
+                    new PackageSource("https://feed1"),
+                    new PackageSource("https://feed2"),
+                };
+
+                var projectName = "TestProject";
+                var projectPath = Path.Combine(pathContext.SolutionRoot, projectName);
+                var outputPath = Path.Combine(projectPath, "obj");
+                var dependencyBar = new LibraryDependency(
+                    new LibraryRange(
+                        "foo",
+                        null,
+                        LibraryDependencyTarget.All),
+                    LibraryIncludeFlags.All,
+                    LibraryIncludeFlags.All,
+                    new List<NuGetLogCode>(),
+                    autoReferenced: false,
+                    generatePathProperty: false,
+                    versionCentrallyManaged: false,
+                    LibraryDependencyReferenceType.Direct,
+                    aliases: null,
+                    versionOverride: null);
+
+                var centralVersionFoo = new CentralPackageVersion("foo", VersionRange.Parse("1.0.0"));
+
+                var tfi = CreateTargetFrameworkInformation(new List<LibraryDependency>() { dependencyBar }, new List<CentralPackageVersion>() { centralVersionFoo });
+                var packageSpec = new PackageSpec(new List<TargetFrameworkInformation>() { tfi })
+                {
+                    FilePath = projectPath,
+                    Name = projectName,
+                    RestoreMetadata = new ProjectRestoreMetadata()
+                    {
+                        ProjectName = projectName,
+                        ProjectUniqueName = projectName,
+                        CentralPackageVersionsEnabled = true,
+                        ProjectStyle = ProjectStyle.PackageReference,
+                        OutputPath = outputPath,
+                        Sources = packageSources
+                    }
+                };
+
+                var logger = new TestLogger();
+
+                PackageSourceMapping packageSourceMappingConfiguration = PackageSourceMapping.GetPackageSourceMapping(NullSettings.Instance);
+
+                if (enablePackageSourceMapping)
+                {
+                    string testConfig = Path.Combine(pathContext.WorkingDirectory, "test.config");
+
+                    SettingsTestUtils.CreateConfigurationFile(testConfig, $@"<?xml version=""1.0"" encoding=""utf-8""?>
```

Comment:
Using just this config probably breaks the redirection of the global packages folder, and likely makes the project "leak" packages into the user's global packages folder. 

Probably best to edit the existing config.  
https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestSettingsContext.cs

You can add helper if there are none.

Reply:
Good call, I'll use the `internal` constructor instead